games/rummy/scenes/PlayScene.ts

```diff
@@ -0,0 +1,1160 @@
+      await this.delay(3000);
+    } else {
+      // meldZoneにカードを置けない場合、手札から1枚捨てる
+      console.log('カードはなかったため、捨てる作業へ');
+      await this.delay(3000);
+      this.cardPutOnDeckFront(this.players[1]);
+      this.resetHandsMove(this.players[1]);
+    }
+
+    // TODO: プレイヤーターンに移行するかどうか判定
+    if (counter === 1) {
+      // ループさせる
+      await this.delay(3000);
+      await this.houseTurn();
+    }
+  }
+
+  // 追加したり、削除した後のreset関数
+  private resetHandsMove(player: RummyPlayer) {
+    if (player.playerType === 'house') {
+      player.hand.forEach((card, i) => {
+        this.time.delayedCall(1000, () => {
+          card.playMoveTween(
+            this.playerHandZones[1].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              i * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[1].y - 50
+          );
+        });
+      });
+    } else {
+      // player
+      player.hand.forEach((card, i) => {
+        this.time.delayedCall(1000, () => {
+          card.playMoveTween(
+            this.playerHandZones[0].x -
+              (GAME.CARD.WIDTH + 10) * 2 +
+              i * (GAME.CARD.WIDTH + 10) -
+              350,
+            this.playerHandZones[0].y + 50
+          );
+        });
+      });
+    }
+  }
+
+  private async cardPutOnDeckFront(player: RummyPlayer) {
+    if (player.playerType === 'player') {
+      this.enableCardDraggable();
+      console.log('select card to deck!!');
+      // dragの移動とhandからの削除まとめて
+      await this.playerHandSelectedCard();
+      this.disableCardDraggable();
+      this.resetHandsMove(this.players[0]);
+
+      await this.delay(2000);
+      // eslint-disable-next-line no-else-return
+    } else {
+      // house
+      const randomIndex = Math.floor(
+        Math.random() * this.players[1].hand.length
+      );
+      const discardedCard =
+        this.players[1].hand[randomIndex];
+      if (discardedCard) {
+        this.children.bringToTop(discardedCard);
+        discardedCard.playMoveTween(
+          this.deckFrontCardX,
+          this.deckFrontCardY
+        );
+        discardedCard.playFlipOverTween();
+        this.deckFront.push(discardedCard);
+        player.removeCardFromHand(discardedCard);
+      }
+    }
+  }
+
+  private playerHandSelectedCard(): Promise<void> {
+    return new Promise<void>((resolve) => {
+      this.input.setDraggable(this.players[0].hand);
+      this.createDropZoneEvent().then(() => {
+        this.input.setDraggable(
+          this.players[0].hand,
+          false
+        );
+        resolve();
+      });
+    });
+  }
+
+  private checkIfCardOverDeck(card: Card) {
+    const deckBounds = this.deckFront[0].getBounds();
+    const cardBounds = card.getBounds();
+
+    return Phaser.Geom.Intersects.RectangleToRectangle(
+      deckBounds,
+      cardBounds
+    );
+  }
+
+  // メルドゾーンにおくカードを選定
+  // eslint-disable-next-line class-methods-use-this
+  private findMeldAbleCard(hands: Card[]) {
```

Comment:
英単語的には、Meldableで1単語です。


```suggestion
  private findMeldableCard(hands: Card[]) {
```